addons/hw_drivers/iot_handlers/drivers/DisplayDriver_L.py

```diff
@@ -74,12 +75,16 @@ def run(self):
     def update_url(self, url=None):
         os.environ['DISPLAY'] = ":0." + self._x_screen
         os.environ['XAUTHORITY'] = '/run/lightdm/pi/xauthority'
-        firefox_env = os.environ.copy()
-        firefox_env['HOME'] = '/tmp/' + self._x_screen
+        browser_env = os.environ.copy()
+        for key in ['HOME', 'XDG_RUNTIME_DIR', 'XDG_CACHE_HOME']:
+            browser_env[key] = '/tmp/' + self._x_screen
         self.url = url or 'http://localhost:8069/point_of_sale/display/' + self.device_identifier
-        new_window = subprocess.call(['xdotool', 'search', '--onlyvisible', '--screen', self._x_screen, '--class', 'Firefox'])
-        subprocess.Popen(['firefox', self.url], env=firefox_env)
-        if new_window:
+
+        # --log-level=3 to avoid useless log messages
+        subprocess.Popen([self.browser, self.url, '--start-fullscreen', '--log-level=3'], env=browser_env)
```

Comment:
you are making the browser a variable but the arguments specific to one, should be `browser_args` or something

Reply:
We are keeping this as it is for now as it is hardcoded, we are refactoring it in another task where we add "variability".